File: apps/polyglot-sample/sample_class.c
typedef struct {
    int value;
} Calculator;

Calculator new_calculator(void) {
    Calculator c = {0};
    return c;
}

int calculator_answer(Calculator* c) {
    return 42;
}

int calculator_add(Calculator* c, int x) {
    c->value = c->value + x;
    return c->value;
}

int answer(void) {
    Calculator calc = new_calculator();
    return calculator_answer(&calc);
}

int main(void) {
    return 0;
}

File: apps/polyglot-sample/sample_class.go
package main

type Calculator struct {
	value int
}

func NewCalculator() Calculator {
	return Calculator{value: 0}
}

func (c *Calculator) Answer() int {
	return 42
}

func (c *Calculator) Add(x int) int {
	c.value = c.value + x
	return c.value
}

func answer() int {
	calc := NewCalculator()
	return calc.Answer()
}

func main() {}
